src/helpers/internal_viewport.ts

```diff
@@ -395,4 +407,28 @@ export class InternalViewport {
       this.getters.getRowDimensions(sheetId, this.top).start -
       this.getters.getRowDimensions(sheetId, this.boundaries.top).start;
   }
+
+  /**
+   * This function is used to prevent performance issue
+   * when there are too many rows or cols
+   * and users navigate sheet using CTRL+SHIFT+ArrowUp/ArrowDown
+   */
+  private getNextValue(currentValue: number, targetValue: number, sign: 1 | -1) {
```

Comment:
Still looks a bit like magic to me and i'd need some time to convince myself but still,
I imagine 2 scenarios here:
1.  it works completely - which means we support the  limit cases of diff = 100001, 50001, 5001 etc 
     in which case I think it's pretty much equivalent of saying that if `diff` > 1, we can make a jump of diff - 1` directly
2. we actually have to rely on the fact that we make iterative steps meaning that the number of steps we make (the value returned here) is important and we could see some caveats

Did you try the edge cases by yourself?
I will try it a bit on my own but I think we can go with it at the moment :)

Reply:
=> does not work if we make steps between 100 and (100 + # of cells visible in that dimension)
for instance put a cell content on row 110 in a standard 1080 screen then from A1 try to go to A110 with Ctrl+ArrowDown,
the cell selected should be a the bottom of the viewport, but it will be in the top middle of the viewport